server/mqtt.go

```diff
@@ -2972,15 +2966,17 @@ func (as *mqttAccountSessionManager) getCachedRetainedMsg(subject string) *mqttR
 	return rm
 }
 
-func (as *mqttAccountSessionManager) setCachedRetainedMsg(subject string, rm *mqttRetainedMsg, onlyReplace bool) {
+func (as *mqttAccountSessionManager) setCachedRetainedMsg(subject string, rm *mqttRetainedMsg, onlyReplace bool, copyBytesToCache bool) {
 	rm.expiresFromCache = time.Now().Add(mqttRetainedCacheTTL)
 	if onlyReplace {
-		if _, ok := as.rmsCache.Load(subject); ok {
-			as.rmsCache.Store(subject, rm)
+		if _, ok := as.rmsCache.Load(subject); !ok {
+			return
 		}
-	} else {
-		as.rmsCache.Store(subject, rm)
 	}
+	if copyBytesToCache {
+		rm.Msg = append([]byte(nil), rm.Msg...)
```

Comment:
Could be replaced with `rm.Msg = copyBytes(rm.Msg)`. I just bench'ed it, and it is generally faster and also allocates only what you need, as opposed to append that may create a bigger capacity than what is copied. Run with size of 5. Notice how the append will use 8 bytes.
```
Benchmark_Ivan_CopyBytes
Benchmark_Ivan_CopyBytes-16    	128584900	         8.200 ns/op	       5 B/op	       1 allocs/op
Benchmark_Ivan_Append
Benchmark_Ivan_Append-16       	74689119	        15.79 ns/op	       8 B/op	       1 allocs/op
```
Run with size of 64.
```
Benchmark_Ivan_CopyBytes
Benchmark_Ivan_CopyBytes-16    	57017350	        18.97 ns/op	      64 B/op	       1 allocs/op
Benchmark_Ivan_Append
Benchmark_Ivan_Append-16       	48468579	        24.95 ns/op	      64 B/op	       1 allocs/op
```